### ISSUE TITLE
[Overblog\GraphQLBundle\Annotation\Arg] The default is always a string. 

### ISSUE THREAD

**Issue by sratefiniaina** [overblog/GraphQLBundle#722]
Q | A
-- | --
Bug report? | yes
Feature request? | no
BC Break report? | no
RFC? | no
Version/Branch | 0.12

**Comment by mcg-web** [comment]
Can you provide an example please?

**Comment by sratefiniaina** [comment]
Yes,
In my Query definition i have this arg :
`@GQL\Arg(name="enabled", type="Boolean", default="false")`
But if I do not provide arg from the graphql side, the value is always true.
And when I look at the file var/graqphl/QueryType.php I have this
`'args' => [ [ 'name' => 'enabled', 'type' => Type::boolean(), 'description' => null, 'defaultValue' => 'false',],],`

**Comment by murtukov** [comment]
@sratefiniaina it should be `default=false` (without quotes)

**Comment by sratefiniaina** [comment]
@mcg-web when i do this, value is null and in var/grqphql/QueryType.php :
`'args' => [ [ 'name' => 'enabled', 'type' => Type::boolean(), 'description' => null,],],`

**Comment by Vincz** [comment]
Hey guys! I'll look into this tonight or tomorrow.

**Comment by sratefiniaina** [comment]
Hey, OK thanks

**Comment by murtukov** [comment]
@Vincz, the bug comes from loose comparison, which you btw often use in your code.

The method, that causes the issue is located in `src/Config/Parser/AnnotationParser.php`:
```php
private static function getArgs(array $args = null, \ReflectionMethod $method = null): array
{
    $config = [];
    if ($args && !empty($args)) {
        foreach ($args as $arg) {
            $config[$arg->name] = ['type' => $arg->type]
                + ($arg->description ? ['description' => $arg->description] : [])
                + ($arg->default ? ['defaultValue' => $arg->default] : []);
        }
    } elseif ($method) {
        $config = self::guessArgs($method);
    }

    return $config;
}
```
the `$arg->default` and `$arg->description` are skipped, if value is `false`, `""`, `0` or `null`. 

This condition should be:
```php
+ (null !== $arg->description ? ['description' => $arg->description] : [])
+ (null !== $arg->default ? ['defaultValue' => $arg->default] : []);
```

The file `src/Config/Parser/AnnotationParser.php` is full of loose comparisons.

**Comment by murtukov** [comment]
@sratefiniaina, the bug is fixed. The patched 0.12 will be available in the next release.
